crates/l3/appchain-core-contract-client/src/interfaces/operator.rs

```diff
@@ -0,0 +1,65 @@
+use async_trait::async_trait;
+use common::invoke_contract;
+use common::LocalWalletSignerMiddleware;
+use starknet_accounts::Execution;
+use starknet_core::types::FieldElement;
+use starknet_core::types::StarknetError;
+use std::sync::Arc;
+
+// Operator accepts LocalWalletSignerMiddleware as argument
+pub struct Operator {
+    client: Arc<LocalWalletSignerMiddleware>,
+    address: FieldElement,
+}
+
+impl Operator {
+    pub fn new(address: FieldElement, client: Arc<LocalWalletSignerMiddleware>) -> Self {
+        Self {
+            client: client,
+            address: address,
+        }
+    }
+    pub async fn register_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<Option<Execution<LocalWalletSignerMiddleware>>, StarknetError> {
+        let execution = invoke_contract(
+            &self.client,
+            self.address,
+            "register_operator",
+            vec![new_operator.into()],
+        )
+        .await;
+        Ok(Some(execution))
+    }
+}
+
+#[async_trait]
+pub trait OperatorTrait {
```

Comment:
hey, now that I see it, I think we can skip having traits at all. basically, in ethers rs , we needed traits to abstract the functioning of ethers rs because the abigen! implemented some methods but they could not be called directly. we had to abstract them using our traits. however, here, since we don't have abigen!, we can do the abstraction in our methods itself.

sorry, i should've caught this earlier, but let's remove the traits for now to save time later on. you can just remove the trait and rest of the code should remain the same I believe.